admin-dev/themes/new-theme/js/pages/currency/form/currency-form.js

```diff
@@ -32,19 +38,58 @@ export default class CurrencyForm {
   constructor(currencyFormMap) {
     this.map = currencyFormMap;
     this.$currencyForm = $(this.map.currencyForm);
-    this.getCLDRDataUrl = this.$currencyForm.data('get-cldr-data');
+    this.$currencyFormFooter = $(this.map.currencyFormFooter);
+    this.apiReferenceUrl = this.$currencyForm.data('reference-url');
+    this.referenceCurrencyResource = Vue.resource(this.apiReferenceUrl);
+    this.originalLanguages = this.$currencyForm.data('languages');
+    this.translations = this.$currencyForm.data('translations');
     this.$currencySelector = $(this.map.currencySelector);
     this.$isUnofficialCheckbox = $(this.map.isUnofficialCheckbox);
     this.$isoCodeInput = $(this.map.isoCodeInput);
     this.$exchangeRateInput = $(this.map.exchangeRateInput);
     this.$precisionInput = $(this.map.precisionInput);
     this.$resetDefaultSettingsButton = $(this.map.resetDefaultSettingsInput);
     this.$loadingDataModal = $(this.map.loadingDataModal);
+    this.currencyFormatterId = this.map.currencyFormatter.replace('#', '');
   }
 
   init() {
     this._initListeners();
     this._initFields();
+    this._initState();
+    this._initCurrencyFormatter();
+  }
+
+  _initState() {
+    this.state = {
+      currencyData: this._getCurrencyDataFromForm(),
+      languages: JSON.parse(JSON.stringify(this.originalLanguages)),
```

Comment:
Why convert it to string, and again to json? :thinking:

Reply:
`Object.assign({}, this.originalLanguages);` is maybe enough here :thinking:

Reply:
Nice, I didn't think about it

Reply:
Ohh, except it doesn't work here because it is an array of  objects, but I think this is the solution:
```js
languages: [...this. originalLanguages],
```

Reply:
yeah, spread operator will populate languages with this.originalLanguages props

Reply:
But not a reference to `this.originalLanguages` right? it's a copy?